cellarium/ml/cli.py

```diff
@@ -624,6 +624,22 @@ def main(args: ArgsType = None) -> None:
     if model_name not in REGISTERED_MODELS:
         raise ValueError(f"'model_name' must be one of {list(REGISTERED_MODELS.keys())}. Got '{model_name}'")
     model_cli = REGISTERED_MODELS[model_name]
+
+    # issue a UserWarning if the subcommand is predict and return_predictions is not set to False
+    if isinstance(args, list):
+        subcommand = args[0]
+        return_predictions = None  # will always issue UserWarning
+    else:
+        subcommand = args.get("subcommand", None)
+        return_predictions = args.get("predict", {}).get("return_predictions", None)
+    if (subcommand == "predict") and (return_predictions != "false"):
+        warnings.warn(
+            "The `return_predictions` argument should be set to 'false' when running predict to avoid OOM. "
+            "This can be set at indent level 0 in the config file. Example:\n"
+            "model:  ...\ndata:  ...\ntrainer:  ...\nreturn_predictions: false",
+            UserWarning,
+        )
```

Comment:
I think this might not work if the config file used. Config file is parsed later on in the init method of `LightningCLI`. Probably this  logic needs to be moved somewhere there. It should happen after this line [         self.parse_arguments(self.parser, args)](https://github.com/Lightning-AI/pytorch-lightning/blob/a944e7744e57a5a2c13f3c73b9735edf2f71e329/src/lightning/pytorch/cli.py#L384). This hook https://github.com/Lightning-AI/pytorch-lightning/blob/a944e7744e57a5a2c13f3c73b9735edf2f71e329/src/lightning/pytorch/cli.py#L554 might be a good place.